### PR TITLE
Feature/cbd import api query

### PR DESCRIPTION
**Pull request by sergiomarrocoli**
Sets up the importer and Commitment class for the import checking pages

- adds a field for the commitment source (csv, form, and cbd)
- Adds a field for cbd_id and cfn_approved

Testing:
in console run ```Commitment.destroy_all```
in bash terminal run ```rake import:commitments```
check csv commitments commitment_source is 'csv' and cfn_aproved is true
check cbd imported commitments have commitment source of 'cbd' and cfn_approved nil

### CODE REVIEW

File: app/models/services/cbd_commitment_hash.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class Services::CbdCommitmentHash
+  def initialize(cbd_hash, commitment)
+    @cbd_hash = cbd_hash
+    @commitment = commitment
+  end
+
+  def call
+    cbd_action = @cbd_hash["action"]
+    country = @cbd_hash["contacts"][0]["country"].present? ? Country.find_by(iso: @cbd_hash["contacts"][0]["country"]["identifier"].upcase)
+        : Country.find_by(iso: "--")
```
Comment (yndajas) [inline app/models/services/cbd_commitment_hash.rb:10]: This is doing quite a bit in a ternary. Would it be clearer in a standard `if`, possibly with some variables to make it clear what your accessing (e.g. why the first contact - is that meaningful?)

File: app/models/services/cbd_commitment_hash.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    {
+      cbd_id: @cbd_hash["_id"],
+      name: cbd_action["name"]["en"],
+      description: cbd_action["description"]["en"],
+      country_ids: [country.id],
+      committed_year: @cbd_hash["meta"]["createdOn"].to_date.year,
```
Comment (yndajas) [inline app/models/services/cbd_commitment_hash.rb:17]: Here and above, is it all worth using `dig` for these heavily nested data?

**Comment by yndajas** [comment]
Managed to get the Rails console working again, but the import is still failing. Looks like it's failing on this line because `country` is nil: `country_ids: [country.id],`

From this:
```
country = if !cbd_com['contacts'][0]['country'].nil?
                    Country.find_by(iso: cbd_com['contacts'][0]['country']['identifier'].upcase)
                  else
                    Country.find_by(iso: '--')
                  end
```

I'm hitting the `else` (the condition is nil), but `Country.find_by(iso: '--')` is also nil, and then you're looking for the ID of that nil record further down

File: app/controllers/commitments_controller.rb [hunk 3]
```diff
@@ -55,7 +55,7 @@ def new
   end
 
   def create
-    @commitment = Commitment.new(commitment_params.merge(user: current_user, user_created: true))
+    @commitment = Commitment.new(commitment_params.merge(user: current_user, source: 'form'))
```
[the remark this answers is not in the document]
Comment (sergiomarrocoli) [reply app/controllers/commitments_controller.rb:58]: yup

File: app/models/services/cbd_commitment_hash.rb (hunk 1 above)
Comment (sergiomarrocoli) [reply app/models/services/cbd_commitment_hash.rb:10]: i don't know, we've just inherited it. the API call has been sent to the DBD team, waiting on a response on whether that's the right place to get it and how to get the manager info

File: app/models/services/cbd_commitment_hash.rb (hunk 2 above)
Comment (sergiomarrocoli) [reply app/models/services/cbd_commitment_hash.rb:17]: changed

**Comment by sergiomarrocoli** [comment]
> Managed to get the Rails console working again, but the import is still failing. Looks like it's failing on this line because `country` is nil: `country_ids: [country.id],`
> 
> From this:
> 
> ```
> country = if !cbd_com['contacts'][0]['country'].nil?
>                     Country.find_by(iso: cbd_com['contacts'][0]['country']['identifier'].upcase)
>                   else
>                     Country.find_by(iso: '--')
>                   end
> ```
> 
> I'm hitting the `else` (the condition is nil), but `Country.find_by(iso: '--')` is also nil, and then you're looking for the ID of that nil record further down

should have been '--' not '---'. how did i even do that?

File: app/models/services/cbd_commitment_hash.rb (hunk 1 above)
Comment (sergiomarrocoli) [reply app/models/services/cbd_commitment_hash.rb]: have refactored it a bit and made a note

**Comment by sergiomarrocoli** [comment]
should be working now. i should also add some spec, will do first thing tomorrow

File: app/controllers/commitments_controller.rb [hunk 4]
```diff
@@ -18,8 +18,9 @@ def index
       filter_params[:filters] = []
       filter_params[:filters] << params[:country_filters]
     end
+
     @paginatedCommitments = Commitment.paginate_commitments(filter_params.to_json).to_json
-    @commitments_count = Commitment.where(state: 'live').count
+    @commitments_count = Commitment.published.count
```
Comment (yndajas) [inline app/controllers/commitments_controller.rb:23]: Should apply this refactor to line 8 of the country_commitments_controller too

File: app/models/services/cbd_commitment_hash.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+class Services::CbdCommitmentHash
```
Comment (yndajas) [inline app/models/services/cbd_commitment_hash.rb:1]: Looks great!

**Review by yndajas — CHANGES_REQUESTED** [review]
Just one change to apply to another case, but otherwise looks good

File: app/controllers/commitments_controller.rb (hunk 4 above)
Comment (sergiomarrocoli) [reply app/controllers/commitments_controller.rb:23]: changed

**Review by yndajas — APPROVED** [review]

**Pull request merged**